### PR TITLE
Don't crash if there are no ants or corr products

### PR DESCRIPTION
**Pull request by ludwigschwardt**
This fixes places in DataSet that would crash if the list of antennas
(or correlation products) is empty. The last place affects printing
of the data set and the rest are extraction of sensor values that
should be empty arrays.

### CODE REVIEW

File: katdal/dataset.py [hunk 1]
```diff
@@ -1018,14 +1019,29 @@ def lst(self):
...
+        return np.column_stack([sensor_data(ant.name) for ant in self.ants]) \
+            if self.ants else np.zeros((self.shape[0], 0))
+
+    def _sensor_per_corrprod(self, base_name):
+        """Extract a single sensor per corrprod and safely stack the results."""
+        def sensor_data(antA, antB):
+            return self.sensor['Antennas/%s/%s_%s' % (antA, base_name, antB)]
+        return np.column_stack([sensor_data(inpA[:-1], inpB[:-1])
+                                for inpA, inpB in self.corr_products]) \
+            if self.corr_products else np.zeros((self.shape[0], 0))
```
Comment (spassmoor) [inline katdal/dataset.py]: This line is returning the error 

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

when there are core-prods and you are requesting    the u,v,w parameters 
e.g   h5.u

Comment (ludwigschwardt) [inline katdal/dataset.py]: Thank you for picking this up. Of course I only checked on the problematic corner case and not the normal files... I really need some unit tests here!
